src/transformers/models/oneformer/image_processing_oneformer.py

```diff
@@ -473,26 +470,30 @@ def __init__(
         self.repo_path = repo_path
         self.metadata = prepare_metadata(load_metadata(repo_path, class_info_file))
         self.num_text = num_text
-        self._valid_processor_keys = [
-            "images",
-            "task_inputs",
-            "segmentation_maps",
-            "instance_id_to_semantic_id",
-            "do_resize",
-            "size",
-            "resample",
-            "do_rescale",
-            "rescale_factor",
-            "do_normalize",
-            "image_mean",
-            "image_std",
-            "ignore_index",
-            "do_reduce_labels",
-            "return_tensors",
-            "data_format",
-            "input_data_format",
-        ]
+        self.num_labels = num_labels
 
+    @classmethod
+    def from_dict(cls, image_processor_dict: Dict[str, Any], **kwargs):
+        """
+        Overrides the `from_dict` method from the base class to save support of deprecated `reduce_labels` in old configs
+        """
+        image_processor_dict = image_processor_dict.copy()
+        if "reduce_labels" in image_processor_dict:
+            image_processor_dict["do_reduce_labels"] = image_processor_dict.pop("reduce_labels")
+        return super().from_dict(image_processor_dict, **kwargs)
```

Comment:
this is great, is there a way to abstract this to a generic method too?

Reply:
I would prefer to remove class specific `from_dict` method since now the decorator handling this, but for that, we have to update `super().from_dict`, which is currently separate `image_processor_dict` and `**kwargs` assignment for image processor initialization.

 - `image_processor_dict` is passed to `__init__` of image processor
 - `**kwargs` are assigned to image_processor instance with `image_processor.key = value`
 
 I addressed it in https://github.com/huggingface/transformers/pull/30799/commits/2b21c88f65e41b6b971afb77e4376b0c98d26d19, however, there would be a small breaking change:

Previously:
 - ✅ `**kwargs` have higher priority over `image_processor_dict`.
 - 🔴 deprecated kwarg in `**kwargs` has higher priority then **not** deprecated one in `image_processor_dict`

Updated: 
 - ✅ `**kwargs` have higher priority over `image_processor_dict`.
 - 🔴 **not** deprecated argument has higher priority no matter where it was specified either in `image_processor_dict` or in `**kwargs`
 
 For example:
 
 ```python
 image_processor_dict = {"do_reduce_labels": True}

 # Before
image_processor = MaskFormerImageProcessor.from_dict(image_processor_dict, reduce_labels=False)
image_processor.do_reduce_labels == False  # reduce_labels deprecated but still has higher priority
 
 # Now
image_processor = MaskFormerImageProcessor.from_dict(image_processor_dict, reduce_labels=False)

# reduce_labels deprecated -> do_reduce_labels from `image_processor_dict` has higher priority 
# + WARNING will be issued
image_processor.do_reduce_labels == True  


 ```

Reply:
I see! thanks for the clear explanation. Then, I think it's ok to have a class-specific method :)